src/backend/mailbox_manager/tasks.py

```diff
@@ -0,0 +1,54 @@
+"""Mailbox manager tasks."""
+
+import logging
+
+import requests
+from celery.schedules import crontab
+
+from mailbox_manager.models import MailDomain, MailDomainStatusChoices
+from mailbox_manager.utils.dimail import DimailAPIClient
+from people.celery_app import app as celery_app
+
+logger = logging.getLogger(__name__)
+
+
+@celery_app.on_after_finalize.connect
+def setup_periodic_tasks(sender, **kwargs):
+    """
+    Setup periodic tasks.
+    """
+    sender.add_periodic_task(
+        crontab(minute="*/5"), fetch_domains_status_task.s()
+    )
+
+
+# TODO: add parameters to use this function in management command too
+def fetch_domains_status():
+    """
+    Call dimail to check and update domains status.
+    """
+    client = DimailAPIClient()
+    # do not fetch status of disabled domains
+    domains = MailDomain.objects.exclude(status=MailDomainStatusChoices.DISABLED)
+    update_count, check_count = 0, 0
+    for domain in domains:
+        old_status = domain.status
+        try:
+            client.fetch_domain_status(domain)
+        except requests.exceptions.HTTPError as err:
+            logger.error("Failed to fetch status for domain %s: %s", domain.name, err)
+        else:
+            if old_status != domain.status:
+                update_count += 1
+            else:
+                check_count += 1
+    # todo: add more details to store more information in the database
+    return f"Domains processed: {update_count} updated, {check_count} checked"
```

Comment:
The counting logic is counter-intuitive. A domain is counted as _either_ checked or updated (so you can have 1 updated and 0 checked). What I would expect is that "updated" is a subset of "checked"…

Also, domains that failed verification are not counted at all… and we probably want to report them because the failure logs are going to be hidden in the Celery container's logs, not directly visible here.

I would prefer something like

`f"Domains processed: {same_count} not changed, {update_count} updated, {failure_count} failed, {check_count} checked"`

with the invariant same_count+update_count+failure_count == check_count.